math/src/fft/fft_cooley_tukey.rs

```diff
@@ -50,10 +50,10 @@ pub fn inverse_cooley_tukey<F: IsField>(
 ) -> Vec<FieldElement<F>> {
     let n = evaluations.len();
     let inverse_n = FieldElement::from(n as u64).inv();
-    let inverse_omega = omega.inv();
+    let inverse_omega = omega.inv().unwrap();
```

Comment:
> This function should propagate the error, since the user of the function could send a 0

One question this function should propagate the error or can panic?
```
pub fn inverse_fft<F: IsField + IsTwoAdicField>(
    evaluations: &[FieldElement<F>],
) -> Result<Vec<FieldElement<F>>, FFTError> {
    let omega = F::get_root_of_unity(log2(evaluations.len())?)?;
    Ok(inverse_cooley_tukey(evaluations, omega))
}
```

Reply:
Yes. The criteria is no function can panic, except the std ops, in particular the div. This is the typical behavior of the commonly used div operator "/". We still need checked ops, so people can have a non panicking alternative, but that's another issue.

Here you can add an FFTError, and use both in the inverse and here

Reply:
@MauroToscano Can you review my last commit? Now the function can propagate the error, but I would like to know if the option I did is correct. Because in `inverse_fft` function can return two types of errors `FieldError` or `FFTError`.

Reply:
It's returning an FFTError, so I'm not sure what do you mean. A function can't return two types of errors unless you are boxing them, and I'm not seeing that.

Reply:
I know it is returning a `FFTError`, but inside this error I added the `FieldError` to it, you can check it in this [file](https://github.com/lambdaclass/lambdaworks/blob/642d80f7c89b003ee5ef7a606e113cbc87d2feb5/math/src/fft/errors.rs#L12) 

I did this because `F::get_root_of_unity(log2(evaluations.len())?)?` can return a `FFTError` and `inverse_cooley_tukey(evaluations, omega)` can return a `FieldError`.

Reply:
@MauroToscano a quick review? This has 3 other approvals and is blocked by this comment only.